tensorflow_federated/python/core/impl/executors/federating_executor.py

```diff
@@ -116,6 +117,331 @@ async def compute(self):
+    fns = await asyncio.gather(*[c.create_value(fn, fn_type) for c in children])
+    results = await asyncio.gather(*[
+        c.create_call(f, v) for c, (f, v) in zip(children, list(zip(fns, val)))
+    ])
+    return FederatingExecutorValue(
+        results,
+        computation_types.FederatedType(
+            fn_type.result, val_type.placement, all_equal=all_equal))
+
+  @tracing.trace
+  async def _zip(self, arg, placement, all_equal):
+    self._check_arg_is_anonymous_tuple(arg)
+    py_typecheck.check_type(placement, placement_literals.PlacementLiteral)
+    children = self._get_child_executors(placement)
+    cardinality = len(children)
+    elements = anonymous_tuple.to_elements(arg.internal_representation)
+    for _, v in elements:
+      py_typecheck.check_type(v, list)
+      if len(v) != cardinality:
+        raise RuntimeError('Expected {} items, found {}.'.format(
+            cardinality, len(v)))
+    new_vals = []
+    for idx in range(cardinality):
+      new_vals.append(
+          anonymous_tuple.AnonymousTuple([(k, v[idx]) for k, v in elements]))
+    new_vals = await asyncio.gather(
+        *[c.create_tuple(x) for c, x in zip(children, new_vals)])
+    return FederatingExecutorValue(
+        new_vals,
+        computation_types.FederatedType(
+            computation_types.NamedTupleType((
+                (k, v.member) if k else v.member
+                for k, v in anonymous_tuple.iter_elements(arg.type_signature))),
+            placement,
+            all_equal=all_equal))
+
+
+class CentralizedIntrinsicStrategy(IntrinsicStrategy):
+  """Strategy for parameter server style execution of intrinsics.
+
+  This is the default intrinsic strategy for FederatingExecutor, in which all
+  intrinsics are coordinated and executed between tff.SERVER and tff.CLIENTS.
+  It is commonly known as the parameter server strategy, with tff.SERVER
+  assuming the role of the parameter server.
+  """
+
+  @classmethod
+  def validate_executor_placements(cls, executor_placements):
+    py_typecheck.check_type(executor_placements, dict)
+    for k, v in executor_placements.items():
+      if k is not None:
+        py_typecheck.check_type(k, placement_literals.PlacementLiteral)
+      py_typecheck.check_type(v, (list, executor_base.Executor))
+      if isinstance(v, list):
+        for e in v:
+          py_typecheck.check_type(e, executor_base.Executor)
+      for pl in [None, placement_literals.SERVER]:
+        if pl in executor_placements:
+          ex = executor_placements[pl]
+          if isinstance(ex, list):
+            pl_cardinality = len(ex)
+            if pl_cardinality != 1:
+              raise ValueError(
+                  'Unsupported cardinality for placement "{}": {}.'.format(
+                      pl, pl_cardinality))
+
+  async def federated_value_at_clients(self, arg):
+    return await executor_utils.compute_intrinsic_federated_value(
+        self.executor, arg, placement_literals.CLIENTS)
```

Comment:
there are some naming inconsistencies that need to be fixed (cf `self.federating_executor` elsewhere). please ignore those

Reply:
fixed